### PR TITLE
[MM-33374] - Adding cloud license to dev environment causes sidebar menu to break

### PR DESCRIPTION
**Pull request by AGMETEOR**
#### Summary

Adding cloud license to dev environment and then naviagting out of the system console causes sidebar menu to break because ```subscriptionStats``` are not loaded in the redux state. This also causes error messages about exceeding the free tier limitations not to be thrown in components like ```invitation_modal_members_step``` and ```invitation_modal_guests_step```. The later scenario is when these components are accessed from the ```Town Square``` channel

#### Ticket Link

[MM-33374](https://mattermost.atlassian.net/browse/MM-33374?atlOrigin=eyJpIjoiOGEzMWUwNzQ5ZDNmNDcxNDk3OTNkZDZmMjIyMjYyNGIiLCJwIjoiaiJ9)

### CODE REVIEW

File: components/common/hocs/cloud/with_get_cloud_subcription.test.tsx [hunk 1]
```diff
@@ -53,6 +72,25 @@ describe('/components/common/hocs/with_get_cloud_subcription', () => {
         expect(getCloudSubscriptionSpy).toHaveBeenCalledTimes(0);
     });
 
+    test('should NOT call the getSubscriptionStats when NOT cloud licenced', () => {
```
Comment (marianunez) [inline components/common/hocs/cloud/with_get_cloud_subcription.test.tsx:75]:
```suggestion
    test('should NOT call the getSubscriptionStats when NOT cloud licenced for non-admin', () => {
```

File: components/common/hocs/cloud/with_get_cloud_subcription.tsx [hunk 2]
```diff
@@ -5,26 +5,37 @@ import React, {ComponentType} from 'react';
...
 }
 
 interface UsedHocProps {
-    subscription: Subscription;
+    subscription?: Subscription;
+    subscriptionStats?: SubscriptionStats;
     isCloud: boolean;
     actions: Actions;
-    userIsAdmin: boolean;
+    userIsAdmin?: boolean;
```
Comment (marianunez) [inline components/common/hocs/cloud/with_get_cloud_subcription.tsx:20]: Does optional make sense here for a bool? I would imagine it always defaults to false

File: components/main_menu/main_menu.jsx [hunk 3]
```diff
@@ -119,6 +120,15 @@ class MainMenu extends React.PureComponent {
...
 
+    shouldShowUpgradeModal = () => {
+        const {subscriptionStats, isCloud} = this.props;
+
+        if (subscriptionStats?.is_paid_tier === 'true') { // eslint-disable-line camelcase
+            return false;
+        }
+        return isCloud && subscriptionStats?.remaining_seats <= 0;
+    }
+
```
Comment (marianunez) [inline components/main_menu/main_menu.jsx:131]: I believe this is no longer required given that Pablo introduced this check within the `invitePeopleModal`.  @pablovelezvidal could you confirm if that is in master?

**Review by marianunez — APPROVED** [review]
Thanks @AGMETEOR! Some minor comments below

File: components/main_menu/main_menu.jsx (hunk 3 above)
Comment (pablovelezvidal) [reply components/main_menu/main_menu.jsx:131]: yes, it was recently merged, so merging master should make the trick.

**Comment by pablovelezvidal** [comment]
Great Work @AGMETEOR ! LGTM!

**Review by pablovelezvidal — APPROVED** [review]

**Review by josephbaylon — APPROVED** [review]
LGTM. Tested cloud license on dev environment for both admin and non-admin and sidebar menu works well.

**Pull request merged**